elyra/pipeline/component_parameter.py

```diff
@@ -298,6 +299,100 @@ def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, e
         runtime_processor.add_disable_node_caching(instance=self, execution_object=execution_object, **kwargs)
 
 
+class CustomSharedMemorySize(ElyraProperty):
+    """An ElyraProperty representing shared memory size for a node."""
+
+    applies_to_generic = True  # custom shared mem size applies to generic components
+    applies_to_custom = True  # custom shared mem size applies to custom components
+
+    property_id = KUBERNETES_SHARED_MEM_SIZE
+    property_display_name = "Shared Memory Size"
+    property_description = """Configure a custom shared memory size in
+    gigabytes (10^9 bytes) for the pod that executes a node. A custom
+    value is assigned if the size property value is a number greater than zero."""
+    property_attributes = [
+        PropertyAttribute(
+            attribute_id="size",
+            display_name="Memory Size (GB)",
+            placeholder=0,
+            input_type="number",
+            hidden=False,
+            required=False,
+        ),
+        PropertyAttribute(
+            attribute_id="units",
+            display_name="Units",
+            input_type="string",
+            hidden=True,
+            required=False,
+        ),
+    ]
+
+    default_units = "G"
+
+    def __init__(self, size: str, units: str, **kwargs):
+        self.size = size
+        self.units = units or CustomSharedMemorySize.default_units
+
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        """Build the JSON schema for an Elyra-owned component property"""
+        schema = super().get_schema()
+        schema["properties"]["size"]["minimum"] = 0
+        return schema
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert instance to a dict with relevant class attributes."""
+        dict_repr = {attr.id: getattr(self, attr.id, None) for attr in self.property_attributes}
+        return dict_repr
+
+    def get_key_for_dict_entry(self) -> str:
+        """
+        Given the attribute names in the 'key' property, construct a key
+        based on the attribute values of the instance.
+        """
+        prop_key = ""
+        keys = [attr.id for attr in self.property_attributes if attr.use_in_key]
+        for key_attr in keys:
+            key_part = getattr(self, key_attr)
+            if key_part:
+                prop_key += f"{key_part}:" if key_attr != keys[-1] else key_part
+        return prop_key
+
+    def get_value_for_dict_entry(self) -> str:
+        """Returns the value to be used when constructing a dict from a list of classes."""
+        return self.to_dict()
```

Comment:
```suggestion
        return self.to_dict()
```
Returns a dict but typehints to a str, should this be Any if the return value can be both str and dict?

Reply:
This method is on `ElyraPropertyListItem` and its subclasses, yet `CustomSharedMemorySize` is a subclass of `ElyraProperty` (and sibling to `ElyraPropertyListItem`).  What is going to call `get_value_for_dict_entry()` on `CustomSharedMemorySize` instances?

Same goes for `to_dict()` - this is only defined for classes of `ElyraPropertyListItem`.

Is this class deriving from the correct superclass or can these two methods simply go away?

Should it be a "list item" simply because it consists of two attributes (one of which is currently hidden)?

Reply:
Good catch! After checking this out, I added a commit that addresses two things:

- parent `get_value_for_dict_entry` method needs to return `Union[str, Dict[str Any]]`
- the `CustomSharedMemorySize` class itself doesn't need either `get_value_for_dict_entry` or `get_key_for_dict_entry`, so these methods are removed

Reply:
@kevin-bates ha! we must have hit respond at the same time. You're completely right though, these methods can go away!